table/tables/partition.go

```diff
@@ -677,7 +681,7 @@ func (lp *ForListPruning) buildListColumnsPruner(ctx sessionctx.Context, tblInfo
 			ExprCol:  columns[idx],
 			valueTp:  &colInfo.FieldType,
 			valueMap: make(map[string]ListPartitionLocation),
-			sorted:   btree.New(btreeDegree),
+			sorted:   btree.NewG[*btreeListColumnItem](btreeDegree, lessBtreeListColumnItem),
```

Comment:
My linter told me that type argument `[*btreeListColumnItem]` is not needed.

Reply:
In the go1.19， Many linter has problems when dealing with generic.